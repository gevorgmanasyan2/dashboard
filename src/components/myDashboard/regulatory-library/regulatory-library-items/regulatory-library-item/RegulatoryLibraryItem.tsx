import React, { FC, useState } from 'react'
import {  notification } from 'antd'
import RegulatoryLibraryItemDetailPopup from './regulatory-library-item-detail/RegulatoryLibraryItemDetailPopup'
import './regulatoryLibraryItem.scss'

interface RegulatoryLibraryItemProps {
  id: number
  name: string
  waiverLibraryItems?: object[] 
}

const RegulatoryLibraryItem: FC<RegulatoryLibraryItemProps> = ({
  id,
  name,
  waiverLibraryItems,
}) => {
  const [visible, setVisible] = useState<boolean>(false)

  const openDetailModal = () => {
    setVisible(true)
  }

  const closeDetailModal = () => {
    setVisible(!visible)
  }

  const openNotification = () => {
    notification.warning({
      message: 'No Data',
    })
  }

  return (
    <>
      <div className="regulatory__library__item__container" 
        onClick={waiverLibraryItems ? () => openDetailModal() :()=>openNotification()}>
        <div className="regulatory__library__item__container--id">
          <span>0{id}</span>
        </div>
        <div className="regulatory__library__item__container--name">
          <span>{name}</span>
        </div>
      </div>
      <RegulatoryLibraryItemDetailPopup
        closeDetailModal={closeDetailModal}
        visible={visible}
        waiverLibraryItems={waiverLibraryItems}
      />
    </>
  )
}

export default RegulatoryLibraryItem
