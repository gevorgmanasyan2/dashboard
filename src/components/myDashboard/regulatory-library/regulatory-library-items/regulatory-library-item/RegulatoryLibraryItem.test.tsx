/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import RegulatoryLibraryItem from './RegulatoryLibraryItem'

it('should render MyDashboard component', () => {
  const mockItem = {
    id: 1,
    name: 'test',
    waiverLibraryItems: [{ id: 1, name: 'test', waiverLibraryItems: [] }],
  }
  expect(shallow(<RegulatoryLibraryItem {...mockItem} />)).toMatchSnapshot()
})
