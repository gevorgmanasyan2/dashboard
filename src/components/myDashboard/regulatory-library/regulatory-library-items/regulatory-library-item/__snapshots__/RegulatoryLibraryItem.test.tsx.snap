// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MyDashboard component 1`] = `
<Fragment>
  <div
    className="regulatory__library__item__container"
    onClick={[Function]}
  >
    <div
      className="regulatory__library__item__container--id"
    >
      <span>
        0
        1
      </span>
    </div>
    <div
      className="regulatory__library__item__container--name"
    >
      <span>
        test
      </span>
    </div>
  </div>
  <RegulatoryLibraryItemDetailPopup
    closeDetailModal={[Function]}
    visible={false}
    waiverLibraryItems={
      Array [
        Object {
          "id": 1,
          "name": "test",
          "waiverLibraryItems": Array [],
        },
      ]
    }
  />
</Fragment>
`;
